edb/schema/pointers.py

```diff
@@ -92,6 +92,78 @@ def merge_cardinality(target: Pointer, sources: List[so.Object],
     return current
 
 
+def merge_readonly(target: Pointer, sources: List[so.Object],
+                   field_name: str, *, schema: s_schema.Schema) -> object:
+
+    current = None
+    current_from = None
+
+    target_source = target.get_source(schema)
+
+    # The target field value is only relevant if it is explicit,
+    # otherwise it should be based on the inherited value.
+    current = target.get_explicit_field_value(schema, field_name, None)
+    if current is not None:
+        current_from = target
+
+    for source in list(sources):
+        # ignore abstract pointers
+        if source.generic(schema):
+            continue
+
+        # We want the field value including the default, not just
+        # explicit value.
+        nextval = source.get_field_value(schema, field_name)
+        if nextval is not None:
+            if current is None:
+                current = nextval
+                current_from = source
+            elif current is not nextval:
+                current_from_source = current_from.get_source(schema)
+                source_source = source.get_source(schema)
+
+                tgt_repr = (
+                    f'{target_source.get_displayname(schema)}.'
```

Comment:
Please use `get_verbosename(with_parent=True)` instead of hand-rolling reprs.

Reply:
Cool, I'll update the `readonly` and `cardinality` error messages to use this instead.